hpobench/benchmarks/surrogates/paramnet_benchmark.py

```diff
@@ -0,0 +1,380 @@
+    @AbstractBenchmark._check_fidelity
+    def objective_function(self, configuration: Union[Dict, CS.Configuration],
+                           fidelity: Union[Dict, None] = None,
+                           rng: Union[np.random.RandomState, int, None] = None, **kwargs) -> Dict:
+
+        config_array = self.convert_config_to_array(configuration)
+        lc = self.surrogate_objective.predict(config_array)[0]
+        costs = self.surrogate_costs.predict(config_array)[0]
+
+        # If we can't afford a single epoch, return TODO.
+        if (costs / self.n_epochs) > fidelity['budget']:
+            # TODO: Return random performance here instead
+            y = 1
+            return {'function_value': float(y),
+                    'cost': fidelity['budget'],
+                    'info': {'state': 'Not enough budget'}}
+
+        learning_curves_cost = np.linspace(costs / self.n_epochs, costs, self.n_epochs)
+
+        if fidelity['budget'] < costs:
+            idx = np.where(learning_curves_cost < fidelity['budget'])[0][-1]
+            y = lc[idx]
+            lc = lc[:idx]
+
+        else:
+            # If the budget is larger than the actual runtime, we extrapolate the learning curve
+            t_left = fidelity['budget'] - costs
+            n_epochs = int(t_left / (costs / self.n_epochs))
+            lc = np.append(lc, np.ones(n_epochs) * lc[-1])
+            y = lc[-1]
+
+        return {'function_value': float(y),
+                'cost': fidelity['budget'],
+                'info': {'learning_curve': lc.tolist(),
+                         'observed_epochs': len(lc)}}
+
+    @AbstractBenchmark._configuration_as_dict
+    @AbstractBenchmark._check_configuration
+    @AbstractBenchmark._check_fidelity
+    def objective_function_test(self, configuration: Union[Dict, CS.Configuration],
+                                fidelity: Union[Dict, None] = None, shuffle: bool = False,
+                                rng: Union[np.random.RandomState, int, None] = None, **kwargs) -> Dict:
+
+        assert fidelity['step'] == 50, f'Only querying a result for the 50. epoch is allowed, ' \
+                                       f'but was {fidelity["step"]}.'
+        return self.objective_function(configuration, fidelity={'step': 50}, rng=rng)
+
+    @staticmethod
+    def _get_fidelity_space(dataset: str, seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+        """
+        Creates a ConfigSpace.ConfigurationSpace containing all fidelity parameters for
+        the SupportVector Benchmark
+
+        Fidelities
+        ----------
+        budget float - [0, Max Time in S]
+            Time which is used to train the network
+
+        Parameters
+        ----------
+        seed : int, None
+            Fixing the seed for the ConfigSpace.ConfigurationSpace
+
+        Returns
+        -------
+        ConfigSpace.ConfigurationSpace
+        """
+        budgets = {  # .   (min, max)-budget (seconds) for the different data sets
```

Comment:
Please add a link to the HPBandster Code to know where these numbers come from

Reply:
Ah, okay, if this is also described in the Paper (as in meta-info), then we don't need a link here